llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -1105,6 +1105,37 @@ static void computeKnownBitsFromOperator(const Operator *I,
       break;
     }
 
+    Value *V;
+    // Handle bitcast from floating point to integer.
+    if (match(const_cast<Operator *>(I), m_ElementWiseBitCast(m_Value(V))) &&
+        V->getType()->isFPOrFPVectorTy()) {
+      KnownFPClass Result = computeKnownFPClass(V, fcAllFlags, Depth + 1, Q);
+      if (Result.SignBit) {
+        if (*Result.SignBit)
+          Known.makeNegative();
+        else
+          Known.makeNonNegative();
+      }
+
+      Type *FPType = V->getType()->getScalarType();
+      int MantissaWidth = FPType->getFPMantissaWidth();
+      if (MantissaWidth != -1) {
+        if (Result.isKnownOnly(fcInf)) {
+          Known.Zero.setLowBits(MantissaWidth);
+          Known.One.setBits(MantissaWidth, BitWidth - 1);
```

Comment:
Should we also verify that the fp semantic of `exponentInf()` is all ones of bitwidth `Bitwidth - 1 - MantissaWidth`. Likewise for nan/zero.

Reply:
Would be a bit safer to use APFloat::getInf().bitcastToAPInt()

Reply:
Maybe a better approach in general would be to intersect the `KnownBits(APFloat::<fpclass>().bitToAPInt())` for all the possible fpclasses.

Reply:
I make it IEEE-only now so it should be safe to set knownbits directly.